compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/partitionning.rs

```diff
@@ -55,7 +55,7 @@ fn extract_levelled_block(dag: &unparametrized::Dag) -> Blocks {
             // Block entry point and pre-exit point
             Op::Lut { .. } => (),
             // Connectors
-            Op::UnsafeCast { input, .. } => uf.union(input.0, op_i),
+            Op::UnsafeCast { input, .. } | Op::ChangePartition { input } => uf.union(input.0, op_i),
```

Comment:
I don't think it should be that way. Since we change partition here, we reach a partition boundary.

Reply:
So I would say that here the `ChangePartition` operator should do the same as for Input and Lut.